sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -50,170 +47,193 @@ namespace Azure { namespace Core {
+     * @return `true` if @DateTime instances are not equal, `false` otherwise.
+     */
+    constexpr bool operator!=(const DateTime& other) const { return !(*this == other); }
 
-    /// Compare the chronological order of two @DateTime instances.
-    bool operator<=(const DateTime& dt) const { return this->m_interval <= dt.m_interval; }
+    /**
+     * @brief Check if \p other @DateTime precedes this @DateTime chronologically.
+     * @param other @DateTime to compare with.
+     * @return `true` if the \p other @DateTime precedes this, `false` otherwise.
+     */
+    constexpr bool operator>(const DateTime& other) const { return !(*this <= other); }
 
-    /// Create an interval from milliseconds.
-    static IntervalType FromMilliseconds(unsigned int milliseconds)
+    /**
+     * @brief Check if \p other @DateTime is chronologically after this @DateTime.
+     * @param other @DateTime to compare with.
+     * @return `true` if the \p other @DateTime is chonologically after this @DateTime, `false`
+     * otherwise.
+     */
+    constexpr bool operator<(const DateTime& other) const
     {
-      return milliseconds * TicksPerMillisecond;
+      return this->m_since1601 < other.m_since1601;
     }
 
-    /// Create an interval from seconds.
-    static IntervalType FromSeconds(unsigned int seconds) { return seconds * TicksPerSecond; }
-
-    /// Create an interval from minutes.
-    static IntervalType FromMinutes(unsigned int minutes) { return minutes * TicksPerMinute; }
-
-    /// Create an interval from hours.
-    static IntervalType FromHours(unsigned int hours) { return hours * TicksPerHour; }
+    /**
+     * @brief Check if \p other @DateTime precedes this @DateTime chronologically, or is equal to
+     * it.
+     * @param other @DateTime to compare with.
+     * @return `true` if the \p other @DateTime precedes or is equal to this @DateTime, `false`
+     * otherwise.
+     */
+    constexpr bool operator>=(const DateTime& other) const { return !(*this < other); }
 
-    /// Create an interval from days.
-    static IntervalType FromDays(unsigned int days) { return days * TicksPerDay; }
+    /**
+     * @brief Check if \p other @DateTime is chronologically after or equal to this @DateTime.
+     * @param other @DateTime to compare with.
+     * @return `true` if the \p other @DateTime is chonologically after or equal to this @DateTime,
+     * `false` otherwise.
+     */
+    constexpr bool operator<=(const DateTime& other) const
+    {
+      return (*this == other) || (*this < other);
+    }
 
-    /// Checks whether this instance of @DateTime is initialized.
-    bool IsInitialized() const { return m_interval != 0; }
+    /**
+     * @brief Get this @DateTime representation as a @Duration from the start of the
+     * implementation-defined epoch.
+     * @return @Duration since the start of the implementation-defined epoch.
+     */
+    constexpr explicit operator Duration() const { return m_since1601; }
```

Comment:
Do we need this? Why would someone want to all this?

Reply:
I think you're right, we can remove it. It was a map of the cpprest's Interval() method. It is used in tests, but I don't see customer using it. For the tests, we still can get it by doing `auto duration(dateTime - DateTime(1601))`.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/pull/967